.pre-commit-config.yaml

```diff
@@ -0,0 +1,8 @@
+repos:
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+    - id: fmt
+    - id: cargo-check
+    - id: clippy
```

Comment:
I can understand `fmt`, but pre-commit `cargo-check` and `clippy` are too slow

Reply:
Well, conversely, if you want that check, you can install a hook locally to perform it. It takes a few seconds if you have everything already precompiled, which may not be the case. Also, more importantly, a git commit is expected to be a) fast, and b) infallible. A `clippy` hook breaks that user experience, and I wouldn't want it to be forced on every user of that repo. Even `fmt` is a gray area, I'm not sure if there won't be any weird side effects during complicated rebases.

Naturally, this repo is `nucypher`'s, so you can override me on this. But for the record, I'm firmly -1, and I will certainly disable it on my local machine.

Reply:
> Well, conversely, if you want that check, you can install a hook locally to perform it.

That is kind of what is occurring here. You would need to install pre-commit for the checks to run (`pre-commit install`), without `pre-commit` installed, the checks won't run. 

To me, we are somewhat trying to enforce these anyway with CI actions (although maybe not on an individual commit level - fair), and these are mirrors of that. 

So, if these local commit checks are viewed as excessive by an individual, the workarounds are :
- don't install pre-commit
OR
- use `git commit --no-verify`

which seem reasonable to me. 

I think some people still don't have `pre-commit` installed for `nucypher/nucypher`, either intentionally or unintentionally.